shared/src/main/kotlin/com/egm/stellio/shared/model/ExpandedEntity.kt

```diff
@@ -86,27 +89,47 @@ data class ExpandedEntity(
             this.plus(propertyKey to JsonLdUtils.buildNonReifiedTemporalValue(dateTime))
         else this
 
-    fun filterOnAttributes(includedAttributes: Set<String>): Map<String, Any> {
+    fun filterOnAttributes(includedAttributes: Set<String>, includedDatasetIds: Set<URI>?): Map<String, Any> {
         val inputToMap = { i: ExpandedEntity -> i.members }
-        return filterEntityOnAttributes(this, inputToMap, includedAttributes)
+        return filterEntityOnAttributes(this, inputToMap, includedAttributes, includedDatasetIds)
     }
 
-    private fun <T> filterEntityOnAttributes(
-        input: T,
-        inputToMap: (T) -> Map<String, Any>,
-        includedAttributes: Set<String>
+    private fun filterEntityOnAttributes(
+
+        input: ExpandedEntity,
+        inputToMap: (ExpandedEntity) -> Map<String, Any>,
+        includedAttributes: Set<String>,
+        includedDatasetIds: Set<URI>?,
     ): Map<String, Any> {
-        return if (includedAttributes.isEmpty()) {
-            inputToMap(input)
-        } else {
+        var result = inputToMap(input)
+
+            if (includedAttributes.isEmpty() && includedDatasetIds == null) {
+                result = inputToMap(input)
+            }
+
+        if (includedAttributes.isNotEmpty()) {
             val includedKeys = JSONLD_EXPANDED_ENTITY_CORE_MEMBERS.plus(includedAttributes)
-            inputToMap(input).filterKeys { includedKeys.contains(it) }
+            result = inputToMap(input).filterKeys { includedKeys.contains(it) }
         }
+        if (includedDatasetIds != null && includedDatasetIds.isNotEmpty()) {
```

Comment:
you can use !includedDatasetIds.isNullOrEmpty()

Reply:
https://github.com/stellio-hub/stellio-context-broker/pull/1177#discussion_r1666949548